core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -147,25 +148,27 @@ public synchronized OutputStream createCompressionStream(OutputStream downStream
     },
 
     GZ(COMPRESSION_GZ) {
-      private transient DefaultCodec codec;
+      private transient AtomicReference<DefaultCodec> codec = new AtomicReference<DefaultCodec>();
 
       @Override
-      synchronized CompressionCodec getCodec() {
-        if (codec == null) {
-          codec = new DefaultCodec();
-          codec.setConf(conf);
+      CompressionCodec getCodec() {
+        DefaultCodec resultCodec = codec.get();
+        if (null == resultCodec) {
+          DefaultCodec newCodec = new DefaultCodec();
+          newCodec.setConf(conf);
+          codec.compareAndSet(null, newCodec);
         }
 
-        return codec;
+        return codec.get();
```

Comment:
I was doing too many things at once and tried to account for the case where we didn't win the CAS operation. I could have checked the result of the compareAndSet method and returned codec.get() in that case and avoided the funny logic to account for that ( which is why I did what I did ). Now that I'm home that's all obvious; however, according to what Brian wrote above I may just take a different approach when I have time again..hopefully tomorrow. I think his tact is simpler